src/hooks/useSharedValue.ts

```diff
@@ -11,5 +11,10 @@ export function useSharedValue<T>(initialValue: T): ISharedValue<T> {
   if (ref.current == null) {
     ref.current = Worklets.createSharedValue(initialValue);
   }
+  // Free on unmount
+  useEffect(() => {
+    return ref.current?.dispose();
+  }, []);
```

Comment:
That will immediately free the shared value, I think we only want to run this in the cleanup function of the useEffect hook, right?

Reply:
Absolutely - this looks like the exact reason for having great people doing code reviews! Thanks!

Reply:
Should be fixed now.